src/Security/Authorization/Core/src/DefaultAuthorizationServiceImpl.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Shared;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authorization;
+
+internal sealed class DefaultAuthorizationServiceImpl(
+    IAuthorizationPolicyProvider policyProvider,
+    IAuthorizationHandlerProvider handlers,
+    ILogger<DefaultAuthorizationService> logger,
+    IAuthorizationHandlerContextFactory contextFactory,
+    IAuthorizationEvaluator evaluator,
+    IOptions<AuthorizationOptions> options,
+    AuthorizationMetrics metrics)
+    : DefaultAuthorizationService(policyProvider, handlers, logger, contextFactory, evaluator, options)
+{
+    public override async Task<AuthorizationResult> AuthorizeAsync(ClaimsPrincipal user, object? resource, IEnumerable<IAuthorizationRequirement> requirements)
+    {
+        AuthorizationResult result;
+        try
+        {
+            result = await base.AuthorizeAsync(user, resource, requirements).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            metrics.AuthorizedRequestFailed(policyName: null, ex);
+            throw;
+        }
+
+        metrics.AuthorizedRequestSucceeded(policyName: null, result);
+        return result;
+    }
+
+    public override async Task<AuthorizationResult> AuthorizeAsync(ClaimsPrincipal user, object? resource, string policyName)
```

Comment:
I mentioned it earlier, but passing in a bool for whether there is an authenticated user or not during authorization seems like it would be useful info.

Reply:
Thanks - I just added a new tag indicating whether the user was authenticated.

I also see there's an experimental `user.roles` tag defined [here](https://opentelemetry.io/docs/specs/semconv/attributes-registry/user/#user-roles). Maybe that's useful information to include as well? cc @halter73

Reply:
You can't put that in a metric. The cardinality could be too high.